mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMapperProvider.java

```diff
@@ -111,7 +111,7 @@ public AnnotationMapper createAnnotationMapper() throws MailboxException {
 
     @Override
     public List<Capabilities> getSupportedCapabilities() {
-        return ImmutableList.copyOf(Capabilities.values());
```

Comment:
copyOf does not necessarilly copy (I know name is misleading).

Also we value readability, and old version is way more redable than the new. Finnaly we prefer relying on guava's collection.

So I disagree with this commit.

Reply:
You are right that copyOf() does not make a copy if the collection is already an `ImmutableList`, with [some exceptions](https://github.com/google/guava/wiki/ImmutableCollectionsExplained#copyof-is-smarter-than-you-think). However, in this case, the collection is an `ArrayList`, which means that toArray() is called:

https://github.com/google/guava/blob/e3b8c0b6f07c1017e80d86758191d6898517a546/guava/src/com/google/common/collect/ImmutableList.java#L252..L259

As an alternative, what about using Iterables.concat()? I have added a commit, "Avoid creating mutableDomains in AbstractDomainList.getDomains()" to show what I mean.